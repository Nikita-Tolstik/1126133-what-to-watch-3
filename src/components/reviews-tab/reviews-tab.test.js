import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsTab from './reviews-tab.jsx';
import {Provider} from 'react-redux';
import configureStore from "redux-mock-store";
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const mockComments = [
  {
    id: 1,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  },
  {
    id: 2,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  },
  {
    id: 3,
    userName: `Sophie`,
    rating: 3,
    comment: `Unfortunately we don't have a reliable way.`,
    date: `2020-03-08T18:06:46.267Z`,
  }
];

it(`Render ReviewsTab - with comments`, () => {
  const store = mockStore({
    [NameSpace.COMMENT]: {
      comments: mockComments,
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ReviewsTab />
      </Provider>
  );

  expect(tree).toMatchSnapshot();
});


it(`Render ReviewsTab - without comments`, () => {
  const store = mockStore({
    [NameSpace.COMMENT]: {
      comments: [],
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ReviewsTab />
      </Provider>
  );

  expect(tree).toMatchSnapshot();
});

