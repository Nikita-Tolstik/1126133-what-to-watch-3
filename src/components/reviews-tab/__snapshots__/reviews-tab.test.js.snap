// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ReviewsTab - with comments 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Unfortunately we don't have a reliable way.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            March 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Unfortunately we don't have a reliable way.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            March 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Unfortunately we don't have a reliable way.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            March 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3
      </div>
    </div>
  </div>
</div>
`;

exports[`Render ReviewsTab - without comments 1`] = `
<p
  style={
    Object {
      "color": "#252525",
    }
  }
>
  Sorry. At this moment no comments yet.
</p>
`;
